doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+We will add the wizard type to our `PluginSet`.
+```diff
+- export type PluginSet = { menu: Plugin[]; editor: Plugin[] };
++ export type PluginSet = { menu: Plugin[]; editor: Plugin[], wizard: Plugin[] };
+```
+
+Wizards must have two public static functions called `CanInspect` and `CanCreate`.
+`Core` will call these functions when it tries to figure out which wizard supports
+the given element and action.
+```ts
+type CanInspect: (tagName: string) => boolean
+type CanCreate: (tagName: string) => boolean
+```
+
+##### Requests
+
+We request a wizard through events.
+
+For viewing and/or editing purposes we can request an inspection.  
+The event is called `oscd-wizard-inspection-request` and its details look like this:
+```ts
+type WizardInspectionRequest = WizardRequest & {
+  element: Element,
+}
+```
+
+The other possible request is to create a new element under a parent element.  
+The event is called `oscd-wizard-creation-request` and its details look like this
+```ts
+type WizardCreationRequest = WizardRequest & {
+  parent: Element,
+  tagName?: string,
+}
+```
+
+Both event detail can contain an additional `stuff` for flexibility. `stuff` can be something
+that the wizard defines for additional configurations (e.g.: display it in readonly mode),
+or the editor can provide additional information about the element.
+
+> **⚠️ WARNING:** Please only use `stuff` if there is no other way.
+> It creates a tight coupling with a specific wizard or editor and it cannot be
+> guaranteed that your expectation about the `stuff` will be met!
+
+```ts
+type WizardRequest = {
+	stuff?: unknown
+}
+```
+
+> **ℹ️ INFO:** you can find the up-to-date definitions here: TODO: file link
+
+> **ℹ️ INFO**: More about other alternatives of requests: 
+> [Alternatives for Requests](#alternatives-of-requests)
+
+##### Finding Available Wizards
+
+When `Core` receives one of the above mentioned events it will go and ask the
+wizards if they can handle the element and action.
+It will call:
+- `CanInspect` if the event is `oscd-wizard-inspection-request` 
+- `CanCreate` if the event is `oscd-wizard-creation-request`.
+
+##### Initiation
+
+In case of inspection the wizard would get the element. 
+```html
+<oscd-wizard-type-templates .element={element} .stuff={stuff}></oscd-wizard-type-templates>
+```
+
+If we want to create a new element it will get the parent element and the new tag name it has to create. Something like this:
+```html
+<oscd-wizard-type-templates .parentElement={parentElement} tag-name="DO" .stuff={stuff}></oscd-wizard-type-templates>
+```
+
+
+##### Wizarding Process
+
+This part happens entirely in the wizard plugin. 
+It can modify the XML document or just display it in some user-friendly way.
+
+##### Self Closing
+
+The wizard decides when it is finished and closes itself. 
+
+`Core` has always the possibility to close the wizard. 
+For example when clicking on the backdrop of a dialog.
```

Comment:
As commented above, we'd likely need some API to let core know when the wizard has done its work.